src/flyte/config/_reader.py

```diff
@@ -92,11 +92,11 @@ def read(self, cfg: typing.Optional["ConfigFile"] = None) -> typing.Optional[typ
 
 
 class ConfigFile(object):
-    def __init__(self, location: str):
+    def __init__(self, location: str | pathlib.Path):
```

Comment:
i think we should leave this as str actually... this isn't user facing right?  if it's not user facing we should be prescriptive.